# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: js/loader/ModuleLoadRequest.h
# Commit: 584beaae1d20
# Full Hash: 584beaae1d20fcf1f066f802e6412673028fa4e5
# Author: Yoshi Cheng-Hao Huang <allstars.chh@gmail.com>
# Date: 2025-10-03 20:53 +0000
# Regressor Bug: 1968890
# File Overlap Count: 3
# Overlapping Files: js/loader/ModuleLoadRequest.h, js/loader/ModuleLoadRequest.cpp, js/loader/ModuleLoaderBase.cpp
# Description:
#   Bug 1968890 - Part 5: Add ModuleLoadRequest::SetErroredLoadingImports(). r=jonco
# ==============================================================================

diff -r e2701bd43ce2 -r 584beaae1d20 js/loader/ModuleLoadRequest.h
--- a/js/loader/ModuleLoadRequest.h	Fri Oct 03 20:53:29 2025 +0000
+++ b/js/loader/ModuleLoadRequest.h	Fri Oct 03 20:53:29 2025 +0000
@@ -113,9 +113,19 @@
 
   const Kind mKind;
 
+  void SetErroredLoadingImports() {
+    MOZ_ASSERT(IsDynamicImport());
+    MOZ_ASSERT(IsFetching());
+    mErroredLoadingImports = true;
+  }
+
   // Type of module (JavaScript, JSON)
   const ModuleType mModuleType;
 
+  // A flag (for dynamic import) that indicates the module script is
+  // successfully fetched and compiled, but its dependencies are failed to load.
+  bool mErroredLoadingImports;
+
   // Pointer to the script loader, used to trigger actions when the module load
   // finishes.
   RefPtr<ModuleLoaderBase> mLoader;